core/src/main/scala/org/http4s/parser/ForwardedHeader.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2013-2020 http4s.org
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.http4s.parser
+
+import java.nio.charset.{Charset, StandardCharsets}
+import java.util.Locale
+
+import org.http4s.headers.Forwarded
+import org.http4s.internal.bug
+import org.http4s.internal.parboiled2._
+import org.http4s.internal.parboiled2.support._
+import org.http4s.{ParseResult, Uri => UriModel}
+
+private[parser] trait ForwardedHeader {
+  def FORWARDED(value: String): ParseResult[Forwarded] = new Parser(value).parse
+
+  private class Parser(value: String)
+      extends Http4sHeaderParser[Forwarded](value)
+      with Rfc3986Parser {
+
+    override def charset: Charset = StandardCharsets.ISO_8859_1
+
+    override def entry: Rule1[Forwarded] =
+      rule {
+        oneOrMore(ELEMENT).separatedBy(ListSep) ~ EOL ~> { (elems: Seq[Forwarded.Element]) =>
+          Forwarded(elems.head, elems.tail: _*)
+        }
+      }
+
+    // TODO: remove this override once the original `ListSet` gets fixed.
```

Comment:
I remember a conversation about this.  Are we tracking this somewhere?

Reply:
I guess nope.
The original question I raised is here: https://gitter.im/http4s/http4s-dev?at=5f056255dbf01050ab5e7a14.
I'll create a separate issue for that, if you don't mind.

Reply:
That'd be great.  Sorry I let that one fall through the cracks.

Reply:
Created the issue #3657 (for reference)